README.md

```diff
@@ -7,3 +7,34 @@
 [![Codecov](https://codecov.io/gh/invenia/NamedDims.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/invenia/NamedDims.jl)
 
 `NamedDimsArray` is a zero-cost abstraction to add names to the dimensions of an array.
+
+## Core functionality:
+For `nda = NamedDimsArrays{(:x,:y,:z)}(rand(10,20,30))`.
+
+ - Unwrapping: `parent(nda)`: returns the underlying `AbstractArray` that is wrapped by the `NamedDimsArray`
+ - Indexing: `nda[y=2]`: the same as `nda[x=:, y=2, z=:]` which is the same as `nda[:,2,:]`
+ - Functions taking a dims arg: `sum(nda; dims=:y)` is the same as `sum(nda; dims=2)`
+
+### Dimensionally Safe Operations
+
+Any operation of multiple `NamedDimArray`s must have compatible dimension names.
+For example trying `NamedDimsArray{(:time,)}(ones(5)) + NamedDimsArray{(:place,)}(ones(5))`
+will throw an error.
+If you perform an operation between another `AbstractArray` and a `NamedDimsArray`, then
+the result will take its names from the `NamedDimsArray`.
+You can use this to bypass the protection,
+ e.g. `NamedDimsArray{(:time,)}(ones(5)) + parent(NamedDimsArray{(:place,)}(ones(5)))`
+ is allowed.
+
+### Partially Named Dimensions (`:_`)
+
+To allow for arrays where only some dimensions have names,
+the name `:_` is treated as a wildcard.
```

Comment:
perhaps advise users against assigning the name `:_`, just noting that it will be given automatically to a dimension that "loses" its name due to soem operation?

Reply:
Also on docs, we should mention that names must be unique (at least for now)

Reply:
done